app/models/envirotech/mappable.rb

```diff
@@ -0,0 +1,72 @@
+module Envirotech
+  module Mappable
+    def self.included(klass)
+      klass.settings = {
+        index: {
+          analysis: {
+            analyzer:
+              { english_analyzer:    {
+                tokenizer: 'standard',
+                filter:    %w(standard asciifolding lowercase snowball) },
+                french_analyzer:     {
+                  tokenizer: 'standard',
+                  filter:    %w(standard asciifolding lowercase french) },
+                chinese_analyzer:    {
+                  tokenizer: 'standard',
+                  filter:    %w(standard  lowercase) },
+                portuguese_analyzer: {
+                  tokenizer: 'standard',
+                  filter:    %w(standard asciifolding lowercase portuguese) },
+                spanish_analyzer:    {
+                  tokenizer: 'standard',
+                  filter:    %w(standard asciifolding lowercase spanish) },
+
+              },
+            filter:   {
+              french:     { type:     'snowball',
+                            language: 'French' },
+              portuguese: { type:     'snowball',
+                            language: 'Portuguese' },
+              spanish:    { type:     'snowball',
+                            language: 'Spanish' },
+            },
+          },
+        },
+      }.freeze
+
+      klass.mappings = {
+        klass.name.typeize => {
+          _timestamp: {
+            enabled: true,
+            store:   true,
+          },
+          properties: {
+            name_chinese:        { type: 'string', analyzer: 'chinese_analyzer' },
+            name_english:        { type: 'string', analyzer: 'english_analyzer' },
+            name_french:         { type: 'string', analyzer: 'french_analyzer' },
+            name_portuguese:     { type: 'string', analyzer: 'portuguese_analyzer' },
+            name_spanish:        { type: 'string', analyzer: 'spanish_analyzer' },
+
+            abstract_chinese:    { type: 'string', analyzer: 'chinese_analyzer' },
+            abstract_english:    { type: 'string', analyzer: 'english_analyzer' },
+            abstract_french:     { type: 'string', analyzer: 'french_analyzer' },
+            abstract_portuguese: { type: 'string', analyzer: 'portuguese_analyzer' },
+            abstract_spanish:    { type: 'string', analyzer: 'spanish_analyzer' },
+
+            source_created_at:   { type: 'date', format: 'YYYY-MM-dd' },
+            source_updated_at:   { type: 'date', format: 'YYYY-MM-dd' },
+            source:              { type: 'string', analyzer: 'keyword' },
+            url:                 { type: 'string' },
+            source_id:           { type: 'string' },
+          },
+        },
+      }.freeze
```

Comment:
Yeah I agree, it is worth keeping that in mind for future, but at this stage I only can see texts in several languages that need to be analyzed in the same way.